### PR TITLE
Dev/search in planner (PRA-219)

### PR DESCRIPTION
**Pull request by AnnaPestova1**
## One Line Description
added search to the Planner
## Requirements

_What this code should do_

## Notes

_Any additional info_

## Test Steps

_Instructions on how to test the changes_

## UI/UX Outcome

_Attach a screenshot here_

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed

### CODE REVIEW

**Review by rebekahcallkacz — CHANGES_REQUESTED** [review]
There seems to be a bug - when I filter my recipes and drag one of the filtered recipes onto the planner, a different recipe shows up.

Initial list: [lemon chicken, strawberry smooth, sriracha lime chicken, pb banana smoothie]
Filtered list: [lemon chicken, sriracha lime chicken]

In the initial list, the recipe I want is the 3rd recipe (sriracha lime chicken). In the filtered list, it is the 2nd recipe. When I drag and drop sriracha chicken from the filtered list, the strawberry smoothie (which is the second item in the original list but does not show up in the filtered list) is what shows up in the planner.

Let me know if you can't replicate the bug and need a video!

File: src/utils/types.tsx [hunk 1]
```diff
@@ -141,7 +143,7 @@ export type AnimationVariants = {
   };
 };
 
-export type PlannerRecipe = {
+export interface PlannerRecipe extends SavedRecipe {
```
[the remark this answers is not in the document]
Comment (AnnaPestova1) [reply src/utils/types.tsx:146]: image
yep.

File: src/components/SingleRecipePage/SingleRecipePage.tsx [hunk 2]
```diff
@@ -42,7 +42,8 @@ const SingleRecipePage = () => {
   const [_sendingIngredients, setSendingIngredients] = useState({});
   // const [openModal, setOpenModal] = useState(false);
   const { isOpen: openNutrition, onToggle } = useDisclosure();
-  const { isOpen: openModal, onClose } = useDisclosure();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (AnnaPestova1) [reply src/components/SingleRecipePage/SingleRecipePage.tsx:40]: Thank you! I will remove it

**Review by AnnaSolovykh — APPROVED** [review]

**Comment by AnnaPestova1** [comment]
> There seems to be a bug - when I filter my recipes and drag one of the filtered recipes onto the planner, a different recipe shows up.
> 
> Initial list: [lemon chicken, strawberry smooth, sriracha lime chicken, pb banana smoothie] Filtered list: [lemon chicken, sriracha lime chicken]
> 
> In the initial list, the recipe I want is the 3rd recipe (sriracha lime chicken). In the filtered list, it is the 2nd recipe. When I drag and drop sriracha chicken from the filtered list, the strawberry smoothie (which is the second item in the original list but does not show up in the filtered list) is what shows up in the planner.
> 
> Let me know if you can't replicate the bug and need a video!

I hope I fixed

**Review by rebekahcallkacz — APPROVED** [review]
Tested again and it works now

I successfully:
- added a recipe to the planner
- removed a recipe from the planner
- added a recipe to the planner from a filtered version of the recipes
- cleared all recipes
- added all ingredients to shopping list

Great job!

**Review by rebekahcallkacz — COMMENTED** [review]
I think you can merge this one after it passes the checks and whenever you're ready!

**Pull request merged**